fdbrpc/TLSConnection.actor.cpp

```diff
@@ -71,23 +74,38 @@ static int recv_func(void* ctx, uint8_t* buf, int len) {
 }
 
 ACTOR static Future<Void> handshake( TLSConnection* self ) {
+	if(!self->is_client) {
+		IPAddress peerIP = self->conn->getPeerAddress().ip;
+		auto iter(serverTLSConnectionThrottler.find(peerIP));
+		if(iter != serverTLSConnectionThrottler.end() && now() < iter->second) {
+			TraceEvent("TLSIncomingConnectionThrottlingWarning", self->getDebugID()).suppressFor(1.0).detail("PeerIP", peerIP.toString());
+			wait(delay(FLOW_KNOBS->CONNECTION_MONITOR_TIMEOUT));
+			throw connection_failed();
```

Comment:
Do we want to let the violating user know that they are being throttled? I can imagine a case where they immediately realize the problem with the cert and try to connect using the correct one, but since they are still within the 10 seconds throttling time, they would again get a connection failed error, which in this case can be confusing I guess.

Reply:
The client cannot know that this is happening without changing the protocol. You could also get throttled if a different client on the same machine has a wrong certificate, but yo have to correct one